File: code/backend/multicloud-guardian/http/src/main/kotlin/pt/isel/leic/multicloudguardian/http/model/storage/FileOutputModel.kt
package pt.isel.leic.multicloudguardian.http.model.storage

import pt.isel.leic.multicloudguardian.domain.file.File

data class FileOutputModel(
    val fileId: Int,
    val userId: Int,
    val folderId: Int?,
    val name: String,
    val path: String,
    val size: Int,
    val encryption: Boolean,
    val url: String,
) {
    companion object {
        fun fromDomain(file: File): FileOutputModel =
            FileOutputModel(
                fileId = file.fileId.value,
                userId = file.userId.value,
                folderId = file.folderId?.value,
                name = file.name,
                path = file.path,
                size = file.size,
                encryption = file.encryption,
                url = file.url,
            )
    }
}

File: code/frontend/services/storage/model/FileOutputModel.ts
import { UserInfo } from "@/domain/user/UserInfo";
import { FolderInfoDetailsModel } from "./FolderInfoDetailsModel";

export type FileOutputModel = {
  fileId: number;
  user: UserInfo;
  folderInfo: FolderInfoDetailsModel | null;
  name: string;
  path: string;
  size: number;
  contentType: string;
  createdAt: number;
  encryption: boolean;
  url: string | null;
};
